runtime/jnichk/jnicheck.c

```diff
@@ -548,21 +548,25 @@ jniCheckCallA(const char* function, JNIEnv* env, jobject receiver, UDATA methodT
 void
 jniCheckArgs(const char *function, int exceptionSafe, int criticalSafe, J9JniCheckLocalRefState *refTracking, const U_32 *descriptor, JNIEnv *env, ...)
 {
-	J9JavaVM *vm = ((J9VMThread*)env)->javaVM;
-	PORT_ACCESS_FROM_JAVAVM(vm);
+	PORT_ACCESS_FROM_JAVAVM(globalJavaVM);
 	va_list va;
 	const U_32 *code;
 	int argNum = 2;
-	UDATA trace = vm->checkJNIData.options & JNICHK_TRACE;
-	UDATA warn = (0 == (vm->checkJNIData.options & JNICHK_NOWARN));
-	J9VMThread * vmThread = (J9VMThread*)env;
+	UDATA trace = globalJavaVM->checkJNIData.options & JNICHK_TRACE;
+	UDATA warn = (0 == (globalJavaVM->checkJNIData.options & JNICHK_NOWARN));
+	J9VMThread *vmThread = (J9VMThread*)env;
+	J9VMThread *currentVMThread = globalJavaVM->internalVMFunctions->currentVMThread(globalJavaVM);
 
-	if (vm->reserved1_identifier != (void*)J9VM_IDENTIFIER) {
-		jniCheckFatalErrorNLS(env, J9NLS_JNICHK_INVALID_ENV, function);
+	if (NULL == currentVMThread) {
+		jniCheckFatalErrorNLS((JNIEnv*)currentVMThread, J9NLS_JNICHK_THREAD_NOT_ATTACHED_TO_VM, function);
```

Comment:
Come to think of it, if `currentVMThread` is null then there's no way calling `jniCheckFatalErrorNLS` will ever work properly. Do you think in this case using a lower level print function would make sense?

Reply:
The error function should be using the global VM for anything important (like finding the port library to do the printing), and use the env only as the printed value of the invalid thread.

Reply:
`currentVMThread` is already the thread that comes from the global VM though, so if it's null then there's no other thread to use.

Reply:
I'm saying `jniCheckFatalErrorNLS` needs to be updated to get the port library from the global VM and not from deferencing a possibly NULL thread. All of the warning/error functions should be modified for consistency.

Reply:
Essentially, every `->javaVM` reference that is not preceded by the arg check needs to either be replaced with the global.

Reply:
That would be fine.

Reply:
Ok, I removed the `env` parameter from those functions in [810142c](https://github.com/eclipse-openj9/openj9/commit/810142cf74341d58e1e1cca7aef7f1927afc364d).